### ISSUE TITLE
Race during loading external dictionaries after CH restart

### ISSUE THREAD

**Issue by Jokser** [ClickHouse/ClickHouse#12665]
CH version is 20.5.2

Dictionaries config should be like this: `<dictionaries_config>dictionaries.d/*.xml</dictionaries_config>` 
Create an external dictionary via xml in `dictionaries.d `directory.
Create a table associated with a dictionary: 
```
ATTACH TABLE dict_global_senders
(
    id UInt64,
    ... fields ...
) ENGINE = Dictionary('dict_global_senders')
```
Name in Dictionary clause should be in single quotes ('')
Ensure metadata file with a dictionary is on the filesystem: .../metadata/dictionaries/dict_global_senders.sql
Stop and start ClickHouse.
ClickHouse startup will be failed with the exception like this: 
```
2020.07.22 13:24:03.469592 [ 1529 ] {} <Error> Application: Caught exception while loading metadata: Code: 36, e.displayText() = DB::Exception: external dictionary 'dict_global_senders' not found: Cannot attach table dictionaries.dict_global_senders from metadata file /var/lib/clickhouse/metadata/dictionaries/dict_global_senders.sql from query ATTACH TABLE dict_global_senders (`sender` String, `carrier_id` UInt16, `modified_time` DateTime) ENGINE = Dictionary('dict_global_senders'), Stack trace (when copying this message, always include the lines below):
```

**Comment by den-crane** [comment]
It's broken in 20.4 and 20.5 you can use 20.3.
it has been fixed in testing 20.6.1.4050+

Duplicate https://github.com/ClickHouse/ClickHouse/issues/12454 https://github.com/ClickHouse/ClickHouse/issues/10994

**Comment by den-crane** [comment]
And probably it solved in 20.5.3.27 (just released)
